runelite-client/src/main/java/net/runelite/client/plugins/twitch/TwitchPlugin.java

```diff
@@ -226,4 +295,111 @@ public boolean onPrivateMessageInput(PrivateMessageInput privateMessageInput)
 	{
 		return false;
 	}
+
+
+	@Subscribe
+	private void onMenuOptionClicked(MenuOptionClicked event)
+	{
+		if (event.getWidgetId() != CLAN_BUTTON_ID)
+		{
+			return;
+		}
+
+		if (!event.getMenuOption().equals("Switch tab") && !event.getMenuOption().equals("Switch:") &&
+			!event.getMenuOption().equals(TWITCH_FILTER_MENUOPTION))
+		{
+			return;
+		}
+
+		event.consume();
+
+		clanChatButtonBackground = client.getWidget(WidgetInfo.CHATBOX_CLAN_BUTTON_BACKGROUND);
+		clanChatButtonText = client.getWidget(WidgetInfo.CHATBOX_CLAN_BUTTON_TEXT);
+		clanChatButtonFilterText = client.getWidget(WidgetInfo.CHATBOX_CLAN_BUTTON_FILTER_TEXT);
+
+		if (event.getMenuOption().equals("Switch tab"))
+		{
+			if (twitchChatSelected)
+			{
+				client.runScript(ScriptID.CHAT_BUTTON_ONOP, 1, 6);
```

Comment:
Seems like this script call could be separated into a function that checks `twitchChatSelected` and that method could be used in a majority of places